packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -323,9 +323,28 @@ class EventAggregator implements Observer<ViewSnapshot> {
   }
 }
 
-interface OutstandingWrite {
-  mutations: Mutation[];
-  userCallback: Deferred<void>;
+/**
+ * FIFO queue that tracks all outstanding mutations for a single test run.
+ * As these mutations are shared among the set of active clients, any client can
+ * add or retrieve mutations.
+ */
+// PORTING NOTE: Multi-tab only.
+class OutstandingMutationTracker {
+  private mutations: Mutation[][] = [];
```

Comment:
"mutations" sounds like a list, but you have a list-of-lists.  

Methinks maybe:
1. OutstandingMutationTracker should be OutstandingBatchTracker or OutstandingWriteTracker.
2. "mutations" here should be "batches" or "writes" (and maybe add a comment).
3. push(mutation: Mutation[]) should be push(batch: Mutation[]) or push(write: Mutation[])

Reply:
Renamed to `SharedWriteTracker` and made everything write. I feel the distinction that a write equals multiple mutations could be made more obvious by a typedef... but maybe I'll tackle that another day.